src/common/src/config.rs

```diff
@@ -1039,6 +1028,87 @@ pub struct S3ObjectStoreDeveloperConfig {
     pub use_opendal: bool,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, DefaultFromSerde)]
+pub struct ObjectStoreRetryConfig {
+    #[serde(default = "default::object_store_config::object_store_streaming_read_timeout_ms")]
+    pub streaming_read_timeout_ms: u64,
+    #[serde(default = "default::object_store_config::object_store_streaming_upload_timeout_ms")]
+    pub streaming_upload_timeout_ms: u64,
+    #[serde(default = "default::object_store_config::object_store_upload_timeout_ms")]
+    pub upload_timeout_ms: u64,
+    #[serde(default = "default::object_store_config::object_store_read_timeout_ms")]
+    pub read_timeout_ms: u64,
+
+    #[serde(default = "default::object_store_config::object_store_req_retry_interval_ms")]
+    pub req_retry_interval_ms: u64,
+    #[serde(default = "default::object_store_config::object_store_req_retry_max_delay_ms")]
+    pub req_retry_max_delay_ms: u64,
+    #[serde(default = "default::object_store_config::object_store_req_retry_factor")]
+    pub req_retry_factor: u64,
```

Comment:
Given that these configs are moved under `ObjectStoreRetryConfig` in this PR, backward compat is broken (IMO it is fine since thesse configs are rarely used), can we rename them to `backoff_interval_ms`, `backoff_max_delay_ms` and `backoff_factor`? I find the original naming very confusing and it makes people think they are related to timeout instead of backoff.